railties/test/generators/app_generator_test.rb

```diff
@@ -160,6 +160,12 @@ def test_new_application_doesnt_need_defaults
     assert_no_file "config/initializers/new_framework_defaults_5_2.rb"
   end
 
+  def test_new_5_2_application_has_defaults
+    run_generator
+
+    assert_file "config/initializers/new_framework_defaults_5_2.rb"
+  end
```

Comment:
Newer Rails app `load_defaults` from the version they were created with: https://github.com/rails/rails/blob/e8b4a927340183acba7fd0a2c3864afc2388a9aa/railties/lib/rails/generators/rails/app/templates/config/application.rb#L25-L26

So a brand new 5.2 app shouldn't have this file, it's only for 5.1 upgrading apps.